### PR TITLE
Update DLA format

### PR DESCRIPTION
**Pull request by alxogm**
- Updated the piece of reading DLAs from file to be able to run on London v5 and on. 
- Tested that all DLA ids, goes from transmission to truth file correctly, when no downsampling is applied. Only small differences in the number of DLAs added in quickquasars, due to the conditionals in lines L362 and 372, which I think are ok. 
- Added unique IDs to DLAs added randomly, to match the same format as DLAs from file. 

@londumas @andreufont  Should I use exactly the same names you have agreed for the transmission file in the truth file?  Or was it ok as in the test you used?

### CODE REVIEW

**Comment by andreufont** [comment]
Hi @alxogm, I'll let you and Helion agree with the format.

File: py/desisim/scripts/quickquasars.py [hunk 1]
```diff
@@ -248,8 +248,7 @@ def simulate_one_healpix(ifilename,args,model,obsconditions,decam_and_wise_filte
     # Read transmission from files. It might include DLA information, and it
     # might add metal transmission as well (from the HDU file).
     log.info("Read transmission file {}".format(ifilename))
-    trans_wave, transmission, metadata, dla_info = read_lya_skewers(ifilename,read_dlas=(args.dla=='file'),add_metals=args.metals_from_file)
-
+    trans_wave, transmission, metadata, dla_info = read_lya_skewers(ifilename,read_dlas=(args.dla=='file'),add_metals=args.metals_from_file)
```
Comment (londumas) [inline py/desisim/scripts/quickquasars.py]: What is the change for this line? and why add a space?

File: py/desisim/scripts/quickquasars.py [hunk 2]
```diff
@@ -370,27 +369,31 @@ def simulate_one_healpix(ifilename,args,model,obsconditions,decam_and_wise_filte
...
-                    if dla['Z_DLA']>=metadata['Z'][ii]: continue
-                    dlas.append(dict(z=dla['Z_DLA']+dla['DZ_DLA'],N=dla['N_HI_DLA']))
+                    if dla['Z_DLA_RSD']>=metadata['Z'][ii]: continue
+                    dlas.append(dict(z=dla['Z_DLA_RSD'],N=dla['N_HI_DLA'],dlaid=dla['DLAID']))
                 transmission_dla = dla_spec(trans_wave,dlas)
 
             elif args.dla=='random':
                 dlas, transmission_dla = insert_dlas(trans_wave, metadata['Z'][ii], rstate=random_state_just_for_dlas)
+                for idla in dlas:
+                   idla['dlaid']+=idd*1000      #Added to have unique DLA ids. Same format as DLAs from file.
```
Comment (londumas) [inline py/desisim/scripts/quickquasars.py]: `DLAID` is already unique I think. Why add `idd*1000`?

File: py/desisim/scripts/quickquasars.py [hunk 3]
```diff
@@ -370,27 +369,31 @@ def simulate_one_healpix(ifilename,args,model,obsconditions,decam_and_wise_filte
...
+                    if dla['Z_DLA_RSD']>=metadata['Z'][ii]: continue
+                    dlas.append(dict(z=dla['Z_DLA_RSD'],N=dla['N_HI_DLA'],dlaid=dla['DLAID']))
                 transmission_dla = dla_spec(trans_wave,dlas)
 
             elif args.dla=='random':
                 dlas, transmission_dla = insert_dlas(trans_wave, metadata['Z'][ii], rstate=random_state_just_for_dlas)
+                for idla in dlas:
+                   idla['dlaid']+=idd*1000      #Added to have unique DLA ids. Same format as DLAs from file. 
 
+# Please enter the commit message for your changes. Lines starting
```
Comment (londumas) [inline py/desisim/scripts/quickquasars.py]: Please justify the comment lines with the code.

File: py/desisim/scripts/quickquasars.py (hunk 2 above)
Comment (alxogm) [reply py/desisim/scripts/quickquasars.py]: @londumas This is only for the case when we add DLAs from random, not from transmission file. This is to match the same format as it is now in the case of adding DLAS from transmission file.

File: py/desisim/scripts/quickquasars.py (hunk 1 above)
Comment (alxogm) [reply py/desisim/scripts/quickquasars.py]: No change here, might be a mistake in my editor.

Comment (londumas) [reply py/desisim/scripts/quickquasars.py]: ok, could you revert? Thanks.

**Comment by alxogm** [comment]
@londumas How about the names in the truth file for DLAs? Should I use exactly the same as they come in the transmission file? or it is fine  as it is?  See https://github.com/desihub/desisim/blob/f9e1709dab956bd923ed4dd1d4c70c6ed76b456c/py/desisim/scripts/quickquasars.py#L393-L398

File: py/desisim/scripts/quickquasars.py (hunk 1 above)
Comment (londumas) [reply py/desisim/scripts/quickquasars.py]: @alxogm, it is because you have trailing spaces at the end of the line.

**Comment by londumas** [comment]
@alxogm, tha apparent changes on the line starting with `trans_wave, transmission, metadata, dla_info ` is because you have trailing space at the end of the line.

**Comment by alxogm** [comment]
> @alxogm, tha apparent changes on the line starting with `trans_wave, transmission, metadata, dla_info ` is because you have trailing space at the end of the line.

Yes, I finally saw it. I think all is addressed. Just let me know about the names, and will be done :D thanks

**Comment by londumas** [comment]
Perfect! Thank you for the work @alxogm.

**Comment by alxogm** [comment]
> Perfect! Thank you for the work @alxogm.

Thanks @londumas . If you have no comments on the names in the truth table, then I'll merge when travis tests finish.

**Comment by londumas** [comment]
@alxogm, it is perfect.

**Pull request merged**